ddl/partition.go

```diff
@@ -930,30 +930,10 @@ func onDropTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _
 	return ver, nil
 }
 
-func buildPlacementTruncateRules(rules []*placement.RuleOp, schemaID, tableID, jobID int64, oldIDs []int64, newPartitions []model.PartitionDefinition) []*placement.RuleOp {
-	newRules := make([]*placement.RuleOp, 0, len(oldIDs))
-	for i, oldID := range oldIDs {
-		prefix := fmt.Sprintf("%d_t%d_p%d", schemaID, tableID, oldID)
-		for _, rule := range rules {
-			if strings.HasPrefix(rule.ID, prefix) {
-				// delete the old rule
-				newRules = append(newRules, &placement.RuleOp{
-					Action: placement.RuleOpDel,
-					Rule: &placement.Rule{
-						GroupID: placement.RuleDefaultGroupID,
-						ID:      rule.ID,
-					},
-				})
-
-				// add the new rule
-				rule.Action = placement.RuleOpAdd
-				rule.ID = fmt.Sprintf("%d_t%d_p%d_%s_%d_%d", schemaID, tableID, newPartitions[i].ID, rule.Role, jobID, i)
-				newRules = append(newRules, rule)
-				break
-			}
-		}
-	}
-	return newRules
+func buildPlacementTruncateBundle(oldBundle *placement.Bundle, newID int64) *placement.Bundle {
+	newBundle := oldBundle.Clone()
+	newBundle.ID = placement.GroupID(newID)
```

Comment:
What about Rule.groupID?

Reply:
Fixed.

Reply:
What about Rule.StartKeyHex and EndKeyHex?